ui/src/app/pages/workspace/workspace-edit.tsx

```diff
@@ -202,6 +211,32 @@ export const WorkspaceEditSection = (props) => {
   </div>;
 };
 
+export const WorkspaceText = (props) => {
```

Comment:
If `researchValue` is a prop it shouldn't be changing. This should be a full `React.Component` with props and state, and anything that has the ability to change should be in state.

Reply:
Ohhhhh...yeah and in fact all the logic for warning message and  textColor should happen here as it has direct relation with this Component rather than WorkspaceEdit